src/Bundle/CoreBundle/Controller/DomainController.php

```diff
@@ -177,6 +177,13 @@ public function updateAction(Organization $organization, Domain $domain, Request
     {
         $outOfSyncPersistedConfig = null;
 
+        // check if the main config folder was created, raise an error if the folder could not be created
+        if (!$domainConfigManager->configFolderExists())
```

Comment:
1. This is not the only place, where we should check for the configFolder. In Fact, check for the config folder should be done before all actions in domain config manager. Once place to implement the check and also the auto creation could be method getDomainConfigDir() or in each of: updateConfig(), configExists(), loadConfig(), removeConfig(), listConfig(). 

2. Following the current implementation pattern, you should also not return true or false, but throw an IO Exception. This would already be catched so you should not need to update this controller at all. 

3. The flash message "is missing or could not be created due to directory permission issues." is just a guess. There could also be other reasons. So for the moment, I would just print (as already implemented) the general message "Cannot load config file." The whole exception is also dumped to log, so the user can see there what was the problem

Reply:
you are right, my first idea was to present a more specific message to the user in the interface, but its ok to look up the log files
fixed that